server/lib/src/server/mod.rs

```diff
@@ -1311,6 +1309,11 @@ impl QueryServer {
 
         let key_providers = Arc::new(KeyProviders::default());
 
+        // These needs to be pool_size minus one to always leave a DB ticket
+        // for a writer. But it also needs to be at least one :)
+        debug_assert!(pool_size > 0);
+        let read_ticket_pool = if pool_size - 1 > 0 { pool_size } else { 1 };
```

Comment:
If pool_size is `2` you still get `2` in this case, which doesn't match the comment - suggested code makes this a little easier to understand as well 😄 

```suggestion
        let read_ticket_pool = std::cmp::min(pool_size-1,1);
```

Reply:
Actually wait this will make it 0. It needs to be cmp::max